lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -11,6 +11,7 @@ class DeclarationListener
     def initialize(index, dispatcher, parse_result, file_path)
       @index = index
       @file_path = file_path
+      @visibility_stack = T.let([[:public]], T::Array[T::Array[Symbol]])
```

Comment:
It's for nested classes and modules. You need to remember the stack of visibilities inside each level of nesting in case you have something like this

```ruby
module Foo
  private
  def bar; end

  module Bar
    def baz; end # public!
  end

  def qux; end # private again
end
```

Reply:
Yeah I understand why the stack, what I don't understand is the stack of arrays: `T::Array[T::Array[Symbol]])`

Reply:
Oh, I see what you mean. Yeah, I guess it could just be a flat array indeed.

Reply:
Why can't you just push the inline visibility in the stack?